referenceframe/register.go

```diff
@@ -0,0 +1,35 @@
+package referenceframe
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var registeredFrameImplementers = map[string]reflect.Type{}
+
+// RegisterFrameImplementer allows outside packages to register their implementations of the Frame
+// interface for serialization/deserialization.
+func RegisterFrameImplementer(frame Frame) error {
```

Comment:
why not just add a `FrameType() string` method to Frame and use a named string?

Reply:
Gautham had it written this way and I asked him to remove it.  The Frame interface is something that I would rather avoid adding to unnecessarily and in this case there isn't upside IMO to adding it; it just becomes more boilerplate